src/client/v2/algod/models/types.ts

```diff
@@ -7954,14 +7968,14 @@ export class TealKeyValue implements Encodable {
     if (!this.encodingSchemaValue) {
       this.encodingSchemaValue = new NamedMapSchema([]);
       (this.encodingSchemaValue as NamedMapSchema).pushEntries(
-        { key: 'key', valueSchema: new StringSchema(), omitEmpty: true },
+        { key: 'key', valueSchema: new ByteArraySchema(), omitEmpty: true },
         { key: 'value', valueSchema: TealValue.encodingSchema, omitEmpty: true }
       );
     }
     return this.encodingSchemaValue;
   }
 
-  public key: string;
```

Comment:
Note the only changes not related to the error string being updated are in `TealKeyValue` and `TealValue`